### ISSUE TITLE
Abstract base models to simplify reuse

### ISSUE THREAD

**Issue by brente** [divio/cmsplugin-filer#235]
We're using these plugins a lot. Sometimes we need to extend the models, though, and we have been unable to find a DRY, non hackish way to achieve this, since django CMS plugin models cannot extend other (concrete) plugin models.
The ideal solution would require all definitions to be moved to an abstract class, with a simple concrete class, much like what happens in [djangocms-text-ckeditor](https://github.com/divio/djangocms-text-ckeditor/blob/2.9.1/djangocms_text_ckeditor/models.py).
Impact on existing derived code should be minimal to nil, and the implementation should be trivial, but I'm wondering whether this would require additional tests.
I'm willing to work on this, but a design decision is required first.
Thank you.
Stefano

**Comment by yakky** [comment]
I talked about this with @brente during pycon, I think it worths a chat @divio/django-cms-core

**Comment by ursomniac** [comment]
YES PLEASE!   This capability would be INSANELY helpful for a high-profile project I'm working on!    The native plugins are 90% of everything I could possibly want  (the stakeholders - well...) but there's always a SMALL change I'd like to make.

Teaser: it'd be great to be able to change the "more>>" prompt.
Link: (how is this different than teaser?) - I'm having problems getting the "style" to be extendable
etc.

Having abstract versions of the models would be extremely helpful.
